### DISCUSSION TITLE
Reload the avatars?

### DISCUSSION THREAD

**Discussion by jholder007** [MCLD/greatreadingadventure#1099] [Help!]
I brought in the data from the previous SRP, and I have everything except the avatars. It is trying to display the avatars but all the links are broken...

Is there a way to just reload/add the avatars again? I have the zip, but it does not see it in the private folder anymore. Is there a switch in the code to make it think it needs the avatars again?

![2024-03-10_10-11-01](https://github.com/MCLD/greatreadingadventure/assets/162853392/4d888720-0df4-435d-a1c6-e3e488870683)

**Comment by jholder007** [comment]
I am using a Windows server. Is there a way for me to unzip the avatars and just put them in a folder? Does anyone know the folder path the avatars should be on?

**Comment by k7hpn** [comment]
Okay, this explains some of it. The system isn't designed to use the same database year-over-year, in our program each year is a fresh start and we don't carry over any data (PII or otherwise). The avatar package is a ZIP that is specially configured with a `json` configuration file just for import. When importing the avatars the software copies them into a different directory structure - you won't be able to use the ZIP to easily copy the files into the same directory layout as last year.

You would be able to copy the avatar files from last year's installation into this year's installation. The files should be in these folders:

1. `content\site1\avatarbackgrounds\`
2. `content\site1\avatarbundles\`
3. `content\site1\avatars\`

You can take those directories from last year's installation and copy them to this year's and everything should line up. You can also get last year's badges and other materials from that `site1` directory if you need them.

**Reply by jholder007** [reply]
Thanks for getting back to me; here is the rub: The server that all that info was on died, and I can not access any of those directories. 

I think it will be easiest if I reinstall and create a new DB. The users will all have to re-register and start from zero, which is ok, too.

Thanks!

**Reply by k7hpn** [reply]
D'oh! That's a pain. One other approach I thought of:

1. Clean install in another database
2. Run the avatar import process
3. Capture the avatar directory structure from that install and copy it into your restored-database install

I could probably perform steps 1 and 2 pretty quickly using a Docker container and then upload a ZIP file for you somewhere if you want to give this a go before you start from scratch - just let me know what version of the software you were using (and version of the avatar ZIP if you didn't use the one built in to the release version of the software).

**Reply by jholder007** [reply]
Hmm: interesting idea. So I am not using docker; I have it running on IIS, but I could probably do the same thing? The only problem is this might bring me back to my first issue with the original database: the import of the avatars "Unable to connect to the server to run the job." when importing..

I am running 4.4.1
and the avatars I have are: 4.2.2

Here is what the database is looking for in the avatars:
<img width="1287" alt="2024-03-13_08-34-52" src="https://github.com/MCLD/greatreadingadventure/assets/162853392/2712e174-64bf-4cea-8c3b-c551cbf6e92e">

Thank you so much for your help!

**Reply by k7hpn** [reply]
I ran the import with that version of the software and zipped the three directories that were in the `site1` directory. I've attached it to the [gra-avatars v4.2.2 release](https://github.com/MCLD/gra-avatars/releases/tag/v4.2.2) as `site1.zip`. See if that works for you and let me know!

**Reply by jholder007** [reply]
YOU ARE AWESOME!! a million thanks! they all are working now!. Really you saved me! 
<img width="955" alt="2024-03-13_20-20-31" src="https://github.com/MCLD/greatreadingadventure/assets/162853392/c79e3019-1d3c-4f52-8b07-26897095c319">

**Reply by k7hpn** [reply]
I'm so glad this worked for you, yay!
